src/software/ai/passing/pass_generator.hpp

```diff
@@ -88,6 +88,12 @@ class PassGenerator
      */
     ZonePassMap<ZoneEnum> samplePasses(const World& world);
 
+    /**
+     * Generates a new pass to each zone's centre and rates them to pass it to
+     * thunderscope
+     */
+    void sampleZoneCentrePasses(const World& world);
```

Comment:
We should update the doc for this function to mention visualization. Likewise, it'd be nice to have a more descriptive function name (i.e. more related to sampling a bunch of points for visualization). In fact, we might be able to move this to `cost_function.h` since I dont think it uses anything from this class and it seems more similar to a subset of `ratePass` visualization for the whole field. Though I may have missed something.

Reply:
agreed, maybe this function is more appropriate in cost_vis.py because this function doesn't _contribute to _ pass generation, but uses it

Reply:
I agree. Moved it to `cost_function.h` as that seems to be the most logical place.